### PR TITLE
Add user directories section

### PR DESCRIPTION
**Pull request by alexveecle**
From <https://github.com/nicoburns/blessed-rs/discussions/139>.

### CODE REVIEW

File: data/crates.json [hunk 1]
```diff
@@ -958,6 +958,17 @@
...
+                        {
+                            "name": "User directories",
+                            "recommendations": [{
+                                "name": "dirs",
+                                "notes": "Provide platform-specific locations for configuration, cache, and other data"
+                            }],
+			    "see_also": [{
+				"name": "directories",
+				"notes": "A higher-level library that can also compute paths for applications"
+			    }]
```
Comment (djc) [inline data/crates.json:971]: The indentation looks a bit iffy here, would you mind cleaning that up?

Comment (alexveecle) [reply data/crates.json]: Sorry, damn tabs. Force-pushed a fix.

**Comment by djc** [comment]
So why did you decide to show `dirs` over `directories`? It feels to me like the `directories` API is a bit more modular and higher-level which probably makes more sense for our intended audience.

**Comment by alexveecle** [comment]
From https://github.com/nicoburns/blessed-rs/discussions/139#discussioncomment-10913757

> probably for `dirs` with `directories` mentioned in the notes (HTML links work in notes sections)

But I really don't have a strong opinion. Personally, being able to construct a nice path for my application with `directories` looks nice.

**Comment by nicoburns** [comment]
> From [#139 (reply in thread)](https://github.com/nicoburns/blessed-rs/discussions/139#discussioncomment-10913757)
> 
> > probably for `dirs` with `directories` mentioned in the notes (HTML links work in notes sections)
> 
> But I really don't have a strong opinion. Personally, being able to construct a nice path for my application with `directories` looks nice.

I also don't really have a strong opinion on this. I'd be tempted to list both as "recommendations" (noting that `dirs` is slightly lower-level and `directories` slightly higher level) and stick `etcetera` under "see also".

Tbh, it looks like `dirs` and `directories` crates really ought to be one crate. They're both tiny wrappers around `dirs-sys`.

**Review by nicoburns — APPROVED** [review]

**Comment by djc** [comment]
> I also don't really have a strong opinion on this. I'd be tempted to list both as "recommendations" (noting that `dirs` is slightly lower-level and `directories` slightly higher level) and stick `etcetera` under "see also".

This sounds good to me.

> Tbh, it looks like `dirs` and `directories` crates really ought to be one crate. They're both tiny wrappers around `dirs-sys`.

Agreed.

**Comment by alexveecle** [comment]
Sorry, it's been a busy period. I tried to capture the latest suggestions here. Feel free to just overwrite with any wording you prefer- I'm also happy to apply any suggestions.

**Comment by nicoburns** [comment]
Thanks!

**Pull request merged**